Refs #33328 -- Added some advice regarding handling formset:added/removed in 3rd party libraries

matthiask opened the pull request:
Writing code which works with several versions of Django may be somewhat hard at times. 

I tried (and failed) to rewrite the code in django-content-editor to use `addEventListener` instead of jQuery. However, code using `addEventListener` isn't triggered when using `jQuery.trigger(...)` (that's expected, see https://github.com/jquery/jquery/issues/2476)

I therefore propose this addition to the docs. 

Here's the current version which works both with Django 4.0 and 4.1:
https://github.com/matthiask/django-content-editor/blob/e6b4b8b8dd54b922d1b30b625a38b5e3b765d043/content_editor/static/content_editor/content_editor.js#L508-L516

claudep commented:
Looks good. Pleased to see the "hard work" already done for my task to update django-tinymce :-)

kezabelle commented:
I'd (personally) be inclined to add in an actual code _example_ into this note, to back up what's being said. I can reasonably interpret what's being added, _based on seeing_ your solution in django-content-editor, but I'm not certain I'd grok it enough to produce an equivalent out of the gate, though I'd hopefully get there eventually ;)

I don't think that's a fault of the wording per-se (because I don't know what wording _would_ give me the full picture without just showing me, a picture telling a thousand words) but of the nuance of the many ways to do things in JS, let alone with jQuery in the mix, and differing ideas of events etc.

matthiask commented:
> Looks good. Pleased to see the "hard work" already done for my task to update django-tinymce :-)

Thanks! 

> I'd (personally) be inclined to add in an actual code example into this note, to back up what's being said. I can reasonably interpret what's being added, based on seeing your solution in django-content-editor, but I'm not certain I'd grok it enough to produce an equivalent out of the gate, though I'd hopefully get there eventually ;)

That's a good idea. I added a new subsection at the bottom of the page with an example and removed the addition to the versionchanged note. I think it's strange if the compatibility workaround appears before the description.

I was unsure if I should pass a jQuery object to the `handleFormsetAdded` function or not but I opted not to because it makes the new code look more complicated and I don't like that. I like using vanilla JS a lot 😊

carltongibson reviewed (approved):
Thanks all. Good stuff. 🍰

matthiask commented:
🎉
